include/triton/Dialect/TritonGPU/Transforms/Passes.td

```diff
@@ -169,6 +169,20 @@ def TritonGPUOptimizeThreadLocality : Pass<"tritongpu-optimize-thread-locality",
                            "mlir::triton::TritonDialect"];
 }
 
+def TritonGPUOptimizeGatherLayouts : Pass<"tritongpu-optimize-gather-layouts", "mlir::ModuleOp"> {
```

Comment:
Can you check if it's somehow similar to `TritonGPUOptimizeThreadLocality`?

Reply:
I originally added this logic there, but it seems we prefer adding more granular passes (accelerate matmul is anothoer separate layout picking pass) so I separated the logic out since it's a bit different than the logic for optimizing reduction layouts.

Reply:
Seems like reshape was optimized in `TritonGPUOptimizeThreadLocality`, right? I just thought that the name seems general enough to cover your optimizations. It's up to you

Reply:
> Seems like reshape was optimized in TritonGPUOptimizeThreadLocality, right?

Ah you're right it is, but the docstring doesn't state that! (I'll update it) I'll merge the passes together, since I prefer fewer passes :)

Reply:
Merged them.